view/frontend/web/js/track-checkout.js

```diff
@@ -0,0 +1,68 @@
+/* Track checkout funnel transition in magento's default checkout */
+
+var currentStep = '';
+
+function waitFor(condition, doWhenReady) {
+    var waitForInterval = setInterval(function () {
+        if (condition()) {
+            doWhenReady();
+            clearInterval(waitForInterval);
+        }
+    }, 50);
+}
+
+function trackFunnel(step) {
+    if (step === currentStep) {
+        // prevent duplicate tracking
+        return;
+    }
+    currentStep = step;
+    BoltTrack.recordEvent(step);
+}
+
+function setupListnerForShippingForm() {
```

Comment:
I don’t think this will work with a logged-in customer that has a default shipping address.

Reply:
yes this only handles guest checkout. added todo